### PR TITLE
fix: remove request auth headers from debug logging

### PR DESCRIPTION
**Pull request by JenniferMah**
# Fixes #
Remove request auth header from debugging logs.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-node/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file

### CODE REVIEW

File: lib/base/RequestClient.js [hunk 1]
```diff
@@ -133,7 +133,10 @@ function logRequest(options) {
 
   if (options.headers) {
     console.log('Headers:');
-    console.log(options.headers)
+    const headers = Object.keys(options.headers).filter((header) => {
```
Comment (thinkingserious) [inline lib/base/RequestClient.js:136]: I think we might want to add a test here.

File: lib/base/RequestClient.js [hunk 2]
```diff
@@ -133,7 +133,10 @@ function logRequest(options) {
 
   if (options.headers) {
     console.log('Headers:');
-    console.log(options.headers)
```
Comment (childish-sambino) [inline lib/base/RequestClient.js:136]: Also don't log `options.data` or `response.data`.

File: lib/base/RequestClient.js (hunk 1 above)
Comment (JenniferMah) [reply lib/base/RequestClient.js:136]: Should it be a unit test in the Request.Client.spec? and should the test be comparing the logging output or just checking that `headers` does not contain any auth headers?

Comment (thinkingserious) [reply lib/base/RequestClient.js:136]: I like a unit test in `Request.Client.spec`, minimally checking that `headers` does not contain any forbidden data. Thanks!

File: spec/unit/base/RequestClient.spec.js [hunk 3]
```diff
@@ -46,6 +46,11 @@ describe('lastResponse and lastRequest defined', function() {
     expect(client.lastResponse.body).toEqual('voltron');
   });
 
+  it('should not include request authorization header in filtered headers', function () {
+    const filteredHeaderKeys = client.filterLoggingHeaders(client.lastRequest.headers);
+    expect(filteredHeaderKeys).toEqual(['test-header-key', 'Connection'])
```
Comment (childish-sambino) [inline spec/unit/base/RequestClient.spec.js:51]: There's no authorization header to begin with. Should add this to the test data.

Comment (JenniferMah) [reply spec/unit/base/RequestClient.spec.js:51]: `client.lastRequest.headers` includes the test-header-key, connection, and authorization header when it mocks the request. Is there a additional authorization header that needs to be tested that does not take the format of `Authorization: 'Basic abcd'
` that I need to include in the test data?

Comment (childish-sambino) [reply spec/unit/base/RequestClient.spec.js:51]: Ah, never mind then.

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**